roles/rsyslog/roles/input_roles/journal/defaults/main.yaml

```diff
@@ -0,0 +1,116 @@
+# -----------------------------------
+
+rsyslog_conf_journal_main_modules:
+
+  - name: 'journal_main'
+    type: 'modules'
+    sections:
+
+      - options: |-
+          # Read from journal
+          module(load="imjournal" StateFile="{{ rsyslog_work_dir }}/imjournal.state" RateLimit.Burst="{{ imjournal_ratelimit_burst|default(1000000) }}" RateLimit.Interval="{{ imjournal_ratelimit_interval|default(10) }}" PersistStateInterval="{{ imjournal_persist_state_interval|default(1000) }}" )
+
+      - options: |-
+          # Normalize logs
+          module(load="mmnormalize")
+
+rsyslog_conf_journal_formatting:
+
+  - name: 'journal_formatting'
+    type: 'template'
+    path: '{{rsyslog_config_dir }}'
+    sections:
+
+      - options: |-
+          # formatting
+          template(name="cnvt_to_viaq_timestamp" type="list") {
+              property(name="TIMESTAMP" dateFormat="rfc3339")
+          }
+
+          # rsyslog 8.30.0 and later does case insensitive variable name comparison
+          # which means $!MESSAGE is the same as $!message - HOWEVER - the case
+          # of the variable name is preserved when outputting, so we need to "normalize"
+          # to all lower case so that the internal JSON to string conversion will output
+          # "message" in the outgoing record
+          if $inputname == 'imjournal' then {
+              if strlen($!MESSAGE) > 0 then {
+                  action(type="mmnormalize" ruleBase="{{ rsyslog_config_dir }}/parse_json.rulebase" variable="$!MESSAGE")
+                  # ensure that $!message is set and $!MESSAGE is unset
+                  # if rsyslog is case sensitive, then $!MESSAGE == $!message
+                  if $!MESSAGE == $!message then {
+                      set $.save_message = $!message;
+                      unset $!MESSAGE; # which also unsets $!message
+                      set $!message = $.save_message;
+                      unset $.save_message;
+                  } else {
+                      if strlen($!message) == 0 then {
+                          set $!message = $!MESSAGE;
+                      }
+                      unset $!MESSAGE;
+                  }
+              } else {
+                  if strlen($!log) > 0 then {
+                      action(type="mmnormalize" ruleBase="{{ rsyslog_config_dir }}/parse_json.rulebase" variable="$!log")
+                      if strlen($!message) == 0 then {
+                          set $!message = $!log;
+                      }
+                      unset $!log;
+                  }
+              }
+
+              if strlen($!@timestamp) == 0 then {
+                  if strlen($!time) > 0 then {
+                      set $!@timestamp = $!time; # assumes json-file log timestamp in rfc3339 format already
+                      unset $!time;
+                  } else {
+                      set $!@timestamp = exec_template('cnvt_to_viaq_timestamp');
+                  }
+              }
```

Comment:
@richm  Do you think this is the correct way to set the "@timestamp" ?

Reply:
It is for the viaq/openshift common logging case.  In other cases, you might not need to output a `@timestamp` field (although that is the default time field for Elasticsearch), and you might not have a `time` field (which is specific to docker json-file logs).

Reply:
@pcahyna @richm Do we want to support the docker json-file logs? Are they logged to journal?